service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReturnApplicationAboutToSubmitTest.java

```diff
@@ -30,53 +24,43 @@
         super("return-application");
     }
 
-    @Autowired
-    private Time time;
-
     @Test
     void shouldMigrateSubmittedDocumentToReturnedDocumentBundle() {
-        DocumentReference documentReference = buildDocumentReference();
+        DocumentReference submittedForm = buildSubmittedForm("mockSubmittedForm.pdf");
+
+        ReturnApplication returnApplication = ReturnApplication.builder()
+            .note("Some note")
+            .reason(List.of(INCOMPLETE))
+            .build();
 
-        Map<String, Object> data = ImmutableMap.of(
-            RETURN_APPLICATION, ReturnApplication.builder()
-                .note("Some note")
-                .reason(List.of(INCOMPLETE))
-                .build(),
-            "dateSubmitted", "2050-05-19",
-            "submittedForm", documentReference);
+        CaseData caseData = CaseData.builder()
+            .returnApplication(returnApplication)
+            .dateSubmitted(LocalDate.of(2050,5,19))
+            .submittedForm(submittedForm)
+            .state(State.OPEN)
+            .build();
 
-        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToSubmitEvent(buildCaseDetails(data));
-        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+        CaseData extractedCaseData = extractCaseData(postAboutToSubmitEvent(caseData));
 
-        ReturnApplication returnApplication = caseData.getReturnApplication();
-        DocumentReference returnedDocumentReference = returnApplication.getDocument();
+        returnApplication = returnApplication.toBuilder()
```

Comment:
i would create new variable for it and named it like expectedReturnApplication, reassigning is a bit confusing

Reply:
yeah makes more sense